### PR TITLE
gtest: 1.12.1 -> 1.14.0

### PR DESCRIPTION
**Pull request by simonhollingshead**
## Description of changes

From v1.13.0 onwards, C++14 or later is required.  v1.12.1 supported C++11.

https://github.com/google/googletest/releases/tag/v1.13.0
https://github.com/google/googletest/releases/tag/v1.14.0

As this will require a large rebuild, this change is targeting staging.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2405.section.md
Comment (mweinelt) [inline nixos/doc/manual/release-notes/rl-2405.section.md:102]: I think it would be better to say, which C++ version is now required.

**Comment by mweinelt** [comment]
The release note URLs should go into the commit message.

**Review by lukegb — APPROVED** [review]

**Comment by mweinelt** [comment]
Bumping to the latest upstream commit for rapidjson did not help.

**Comment by lukegb** [comment]
I'll take a peek at rapidjson unless someone beats me to it

**Comment by lukegb** [comment]
#290250 for rapidjson

**Comment by simonhollingshead** [comment]
@vcunat I am proposing a fix for libqtdbustest in #290647.

I have found only two(?) other packages I think are impacted and will also provide fixes for those.

**Comment by simonhollingshead** [comment]
@vcunat You've caught a couple more than me.  The two I have seen fail are net-cpp and (I believe) itpp, but itpp's logs in hydra are missing and just show an error.

**Comment by simonhollingshead** [comment]
I believe that I've now interacted with all x86-64 linux builds where the error message contains the C++14 gtest error message.

All are associated with this PR, above.

Note that in four cases, I've either nudged an existing version bump PR or filed an update request.  These are all cases where upstream have fixed gtest but where there seems to be some complicating factor (e.g. just updating to a new version causes a compile error unrelated to gtest, or one thing can't update until some other thing is ready).  In these cases I feel it's better to defer to maintainers.

**Pull request merged**
